### ISSUE TITLE
Compile error when trying to use std.fmt on u65535

### ISSUE THREAD

**Issue by quentin-k** [ziglang/zig#15937]
### Zig Version

0.11.0-dev.3009+b18b4db70

### Steps to Reproduce and Observed Behavior

Using the following code:
```zig
const std = @import("std");

pub fn main() !void {
    const big_num: u65535 = 1;
    
    const stdout_file = std.io.getStdOut().writer();
    var bw = std.io.bufferedWriter(stdout_file);
    const stdout = bw.writer();

    try stdout.print("{}\n", .{big_num});

    try bw.flush();
}
```
You will get the error of:
```
install transitive failure
└─ install numeric-bug transitive failure
   └─ zig build-exe numeric-bug Debug native 1 errors
$ZIG_PATH/lib/std/fmt.zig:1426:17: error: overflow of integer type 'u16' with value '65536'
    var buf: [1 + math.max(value_info.bits, 1)]u8 = undefined;
              ~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
referenced by:
    formatIntValue__anon_6253: $ZIG_PATH/lib/std/fmt.zig:790:12
    formatValue__anon_6250: $ZIG_PATH/lib/std/fmt.zig:739:38
    remaining reference traces hidden; use '-freference-trace' to see all reference traces
```

### Expected Behavior

I expected a very slow but successful compilation.

**Comment by Rexicon226** [comment]
This can be closed, it no longer reproduces on 0.14.0-dev.1632+d83a3f174.

**Comment by Vexu** [comment]
Fixed in 259315606827620daaabf82b479e59ee710097cd by casting to `comptime_int`.
